sdk/core/core/src/az_span.c

```diff
@@ -146,21 +146,20 @@ AZ_NODISCARD az_result az_span_find(az_span source, az_span target, az_span* out
  */
   AZ_PRECONDITION_VALID_SPAN(source, 1, false);
```

Comment:
I'm fine with that. I'll make those changes.

Reply:
Hey, sorry to walk back on this, but I thinking more about it... the logic discussed in the links relates to strings. I agree any string contains empty strings. But would any byte sequence contain an empty byte sequence (AZ_SPAN_NULL)?
I'm thinking about the value provided by the function. Maybe the results make more sense like
- az_span_find(empty, empty) = 0 // since both are empty, ok target is in source.
- az_span_find(non-empty, empty) = -1 // I'm not sure it is useful to return 0, since AZ_SPAN_NULL would not be part of a non-empty sequence.
- az_span_find(non-empty, non-empty) = n (where n >=0)
- az_span_find(empty, non-empty) = -1

What do you think?

If there is a preference to adhere to how c++ does, I'm fine. We will then have to handle checking if source or target on the calling code to achieve the desired behavior.